lib/revise_auth/authentication.rb

```diff
@@ -52,5 +54,16 @@ def logout
       Current.user = nil
       reset_session
     end
+
+    private
+      def redirect_to_login_with_stashed_location
+        stash_intended_location
+        redirect_to login_path, alert: I18n.t("revise_auth.sign_up_or_login")
+      end
+
+      # Store user intended url, so we can redirect him there after the login.
+      def stash_intended_location
+        session[:user_return_to] = request.original_url if request.get?
```

Comment:
https://github.com/heartcombo/devise/blob/main/lib/devise/controllers/store_location.rb#L56

Wonder if we should strip to just the path as well to be safe?